include/nnstreamer-edge.h

```diff
@@ -96,58 +96,199 @@ typedef void (*nns_edge_data_destroy_cb) (void *data);
 
 /**
  * @brief Create edge handle.
+ * @param[in] id Unique id in local network
+ * @param[in] connect_type value of nns_edge_flag_e. Connection type between edge nodes.
+ * @param[in] flags value of @a nns_edge_flag_e. The value can be set by using bitwise-or operation.
+ * @param[out] edge_h The edge handle. edge handle should be released using nns_edge_release_handle().
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ *
+ * Here is an example of the usage:
+ * @code
+ * static int _nns_edge_event_cb (nns_edge_event_h event_h, void *user_data) {
+ *   nns_edge_event_e event_type;
+ *   nns_edge_event_get_type (event_h, &event_type);
+ *   switch (event_type) {
+ *     case NNS_EDGE_EVENT_CAPABILITY:
+ *     case NNS_EDGE_EVENT_NEW_DATA_RECEIVED:
+ *       // handle event.
+ *       break;
+ *     default:
+ *       break;
+ *   }
+ *   return 0;
+ * }
+ *
+ * // Main function
+ * nns_edge_h edge_h;
+ * int ret;
+ *
+ * // create edge handle.
+ * ret = nns_edge_create_handle ("TEMP_ID", NNS_EDGE_CONNECT_TYPE_TCP, NNS_EDGE_FLAG_RECV | NNS_EDGE_FLAG_SEND | NNS_EDGE_FLAG_SERVER, &edge_h);
+ * if (NNS_EDGE_ERROR_NONE != ret)
+ *   // Handle error case. After this, handling error case is repeated, so it is omitted.
+ *
+ * // [Optional] Set edge info if necessary.
+ * ret = nns_edge_set_info (edge_h, "HOST", "127.0.0.1");
+ *
+ * // Set event callback to receive the new data or capability.
+ * // Query-server, query-client and subscriber need to set event callback.
+ * nns_edge_set_event_callback (src->edge_h, _nns_edge_event_cb, src);
+ *
+ * // start edge handle.
+ * ret = nns_edge_start (edge_h);
+ *
+ * // Do something...
+ *
+ * // Release edge handle.
+ * nns_edge_release_handle (edge_h);
  */
 int nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type, int flags, nns_edge_h *edge_h);
 
 /**
  * @brief Start the nnstreamer edge.
+ * @param[in] edge_h The edge handle.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ * @retval #NNS_EDGE_ERROR_CONNECTION_FAILURE Failed to connect to destination.
```

Comment:
error when failed to set the available port. 
no connection in start()

Reply:
Thanks. Updated.